File: backend/src/main/java/com/hospital/finance/entity/ServiceHospitalier.java
package com.hospital.finance.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Entity
@Table(name = "service_hospitalier")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class ServiceHospitalier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idService;

    @Column(nullable = false)
    private String nomService;

    private Double budgetMensuel;
    private Double budgetAnnuel;
    private Double seuilAlerte;

    // Chef de service (optionnel)
    //@OneToOne(mappedBy = "service")
    //@JsonIgnore   // 🔴 on ne renvoie pas ça dans le JSON
    //private Utilisateur chefService;

    // Relations
    @OneToMany(mappedBy = "service")
    @JsonIgnore
    private List<Sejour> sejours;

    @OneToMany(mappedBy = "service")
    @JsonIgnore
    private List<Prevision> previsions;

    @OneToMany(mappedBy = "service")
    @JsonIgnore
    private List<Alerte> alertes;

    @OneToMany(mappedBy = "service")
    @JsonIgnore
    private List<Consommable> consommables;
}

File: frontend/lib/models/service_hospitalier.dart
class ServiceHospitalier {
  final int? idService;
  final String nomService;
  final double? budgetMensuel;
  final double? budgetAnnuel;
  final double? seuilAlerte;

  ServiceHospitalier({
    this.idService,
    required this.nomService,
    this.budgetMensuel,
    this.budgetAnnuel,
    this.seuilAlerte,
  });

  factory ServiceHospitalier.fromJson(Map<String, dynamic> json) {
    return ServiceHospitalier(
      idService: json['idService'] as int?,
      nomService: json['nomService'] as String,
      budgetMensuel: (json['budgetMensuel'] as num?)?.toDouble(),
      budgetAnnuel: (json['budgetAnnuel'] as num?)?.toDouble(),
      seuilAlerte: (json['seuilAlerte'] as num?)?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idService': idService,
      'nomService': nomService,
      'budgetMensuel': budgetMensuel,
      'budgetAnnuel': budgetAnnuel,
      'seuilAlerte': seuilAlerte,
    };
  }

  ServiceHospitalier copyWith({
    int? idService,
    String? nomService,
    double? budgetMensuel,
    double? budgetAnnuel,
    double? seuilAlerte,
  }) {
    return ServiceHospitalier(
      idService: idService ?? this.idService,
      nomService: nomService ?? this.nomService,
      budgetMensuel: budgetMensuel ?? this.budgetMensuel,
      budgetAnnuel: budgetAnnuel ?? this.budgetAnnuel,
      seuilAlerte: seuilAlerte ?? this.seuilAlerte,
    );
  }
}
